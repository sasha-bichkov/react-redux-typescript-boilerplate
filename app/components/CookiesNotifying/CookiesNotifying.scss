.CookiesNotifying {
  background-color: #0288d1;
  box-shadow:
    0 3px 5px -1px rgb(0 0 0 / .2),
    0 6px 10px 0 rgb(0 0 0 / .14),
    0 1px 18px 0 rgb(0 0 0 / .12);
}

.CookiesNotifying__content {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-between;

  max-width: 1300px;
  margin: 0 auto;
  padding: 6px 16px;
  border-radius: 4px;
}

.CookiesNotifying__leftColumn {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;

  @include mq($from: wide) {
    flex-wrap: nowrap;
    align-items: center;
    justify-content: flex-start;
  }
}

.CookiesNotifying__rightColumn {
  flex-grow: 1;

  text-align: center;
}

.CookiesNotifying__icon {
  width: 30px;

  fill: white;

  @include mq($from: desktop) {
    margin-right: 15px;
  }
}

.CookiesNotifying__text {
  max-width: 980px;
  color: white;
}

.CookiesNotifying__button {
  background-color: #2e7d32;
  color: white;

  &:hover {
    background-color: adjust-color(#2e7d32, $lightness: 10%);
  }

  &:active {
    background-color: adjust-color(#2e7d32, $lightness: -10%);
  }
}
