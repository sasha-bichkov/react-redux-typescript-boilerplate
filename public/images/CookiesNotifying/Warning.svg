<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 30 30"><path d="M15 1.17A13.83 13.83 0 1 0 28.91 15 13.88 13.88 0 0 0 15 1.17Zm0 26.6A12.77 12.77 0 1 1 27.84 15 12.82 12.82 0 0 1 15 27.77Z" stroke="#fff" stroke-miterlimit="10"/><path d="M15 6.23a.55.55 0 0 0-.55.55v12.06a.55.55 0 0 0 1.1 0V6.78a.55.55 0 0 0-.55-.55ZM15 21a.55.55 0 0 0-.55.55v1.09a.55.55 0 0 0 1.1 0v-1.06A.55.55 0 0 0 15 21Z" stroke="#fff" stroke-miterlimit="10"/></svg>
