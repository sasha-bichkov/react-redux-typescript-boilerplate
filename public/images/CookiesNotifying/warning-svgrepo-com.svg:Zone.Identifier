[ZoneTransfer]
ZoneId=3
ReferrerUrl=https://www.svgrepo.com/vectors/warning/
HostUrl=https://www.svgrepo.com/download/36368/warning.svg
