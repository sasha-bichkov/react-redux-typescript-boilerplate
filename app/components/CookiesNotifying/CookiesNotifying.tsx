import React from 'react'

import Button from '@Components/Button'
import WarningIcon from '@Images/CookiesNotifying/Warning.svg'

import './CookiesNotifying.scss'

const CookiesNotifying = () => {
  return (
    <div className="CookiesNotifying">
      <div className="CookiesNotifying__content">
        <div className="CookiesNotifying__leftColumn">
          <WarningIcon className="CookiesNotifying__icon" width="30" />
          <p className="CookiesNotifying__text">
            We&nbsp;use cookies to&nbsp;personalise content and ads,
            to&nbsp;provide social media features and to&nbsp;analyse our traffic.
            We&nbsp;also share information about your use of&nbsp;our site with our social media,
            advertising and analytics partners who may combine it&nbsp;with other information
            that you&rsquo;ve provided to&nbsp;them or&nbsp;that
            they&rsquo;ve collected from your use of&nbsp;their services.
          </p>
        </div>

        <div className="CookiesNotifying__rightColumn">
          <Button className="CookiesNotifying__button">
            accept and close
          </Button>
        </div>
      </div>
    </div>
  )
}

export default CookiesNotifying
