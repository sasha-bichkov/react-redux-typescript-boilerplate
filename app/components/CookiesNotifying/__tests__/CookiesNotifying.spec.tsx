import React from 'react'
import { mount, render, shallow } from 'enzyme'

import CookiesNotifying from '../CookiesNotifying'

describe('CookiesNotifying', () => {
  it('renders correctly', () => {
    const component = shallow(<CookiesNotifying />)

    expect(component).toMatchSnapshot()
  })
  describe('if click button', () => {
    it('calls it when a button is clicked', () => {
      const mockCallback = jest.fn()
      const component = shallow(<CookiesNotifying />)

      console.log(component)
    })
  })
})
