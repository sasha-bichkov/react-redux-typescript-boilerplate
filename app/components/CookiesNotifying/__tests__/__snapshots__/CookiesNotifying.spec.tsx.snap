// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookiesNotifying renders correctly 1`] = `
<div
  className="CookiesNotifying"
>
  <div
    className="CookiesNotifying__content"
  >
    <div
      className="CookiesNotifying__leftColumn"
    >
      <Component
        className="CookiesNotifying__icon"
        width="30"
      />
      <p
        className="CookiesNotifying__text"
      >
        We use cookies to personalise content and ads, to provide social media features and to analyse our traffic. We also share information about your use of our site with our social media, advertising and analytics partners who may combine it with other information that you’ve provided to them or that they’ve collected from your use of their services.
      </p>
    </div>
    <div
      className="CookiesNotifying__rightColumn"
    >
      <Button
        caption="accept and close"
        className="CookiesNotifying__button"
      />
    </div>
  </div>
</div>
`;
